lib/inputstreamhelper/__init__.py

```diff
@@ -132,16 +132,19 @@ def _diskspace(cls):
 
     @classmethod
     def _temp_path(cls):
-        ''' Return temporary path, usually ~/.kodi/userdata/addon_data/tmp '''
-        temp_path = os.path.join(ADDON_PROFILE, 'tmp')
+        ''' Return temporary path, usually ~/.kodi/userdata/addon_data/script.module.inputstreamhelper/temp '''
+        temp_path = ADDON.getSetting('temp_dir')
+        if not temp_path.startswith('special://profile/addon_data/script.module.inputstreamhelper/temp'):
```

Comment:
I'd like to propose one tiny change here:
If the default setting was changed to the path without "temp" at the end, this condition would not be necessary, but rather "temp" always added to the specified path (whether default or custom).

Reply:
I created a new commit. You can revert it if you see a reason.